dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/document/util/DocumentVisibilityUtil.java

```diff
@@ -0,0 +1,316 @@
+        return toMultidimensionalArray(dv);
+    }
+
+    /**
+     * Converts a {@link DocumentVisibility} object into a multidimensional
+     * array representation of the boolean expression.
+     * @param dv the {@link DocumentVisibility}. (not {@code null})
+     * @return the multidimensional array representation of the boolean
+     * expression.
+     */
+    public static Object[] toMultidimensionalArray(final DocumentVisibility dv) {
+        checkNotNull(dv);
+        final byte[] expression = dv.flatten();
+        final DocumentVisibility flattenedDv = DisjunctiveNormalFormConverter.createDnfDocumentVisibility(expression);
+        final Object[] result = toMultidimensionalArray(flattenedDv.getParseTree(), expression);
+        return result;
+    }
+
+    /**
+     * Converts a {@link Node} and its corresponding expression into a
+     * multidimensional array representation of the boolean expression.
+     * @param node the {@link Node}. (not {@code null})
+     * @param expression the expression byte array.
+     * @return the multidimensional array representation of the boolean
+     * expression.
+     */
+    public static Object[] toMultidimensionalArray(final Node node, final byte[] expression) {
+        checkNotNull(node);
+        final List<Object> array = new ArrayList<>();
+
+        if (node.getChildren().isEmpty() && node.getType() == NodeType.TERM) {
+            final String data = getTermNodeData(node);
+            array.add(data);
+        }
+
+        log.trace("Children size: " + node.getChildren().size() + " Type: " + node.getType());
+        for (final Node child : node.getChildren()) {
+            switch (child.getType()) {
+                case EMPTY:
+                case TERM:
+                    String data;
+                    if (child.getType() == NodeType.TERM) {
+                        data = getTermNodeData(child);
+                    } else {
+                        data = "";
+                    }
+                    if (node.getType() == NodeType.OR) {
+                        array.add(Lists.newArrayList(data).toArray(new Object[0]));
+                    } else {
+                        array.add(data);
+                    }
+                    break;
+                case OR:
+                case AND:
+                    array.add(toMultidimensionalArray(child, expression));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        return array.toArray(new Object[0]);
+    }
+
+    public static String nodeToBooleanString(final Node node) {
+        boolean isFirst = true;
+        final StringBuilder sb = new StringBuilder();
+        if (node.getType() == NodeType.TERM) {
+            final String data = getTermNodeData(node);
+            sb.append(data);
+        }
+        if (node.getType() == NodeType.AND) {
+            sb.append("(");
+        }
+        for (final Node child : node.getChildren()) {
```

Comment:
No.  But this is the node's children not the children's children.  We need to handle a node's children.